docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,284 @@
+existing SPL token accounts to ensure compatibility in RPC when fetching pre and
+post token balances.
+
+For a holding, the first 165 bytes must contain the same information
+as a normal SPL token holding. The following byte must be the type of
+account (ie. `Holding`), and after that, any data is allowed as required by the
+new token program.
+
+The same applies for mints, but only for the first 82 bytes.
+
+The `amount` field in an SPL token holding should contain a balance in order
+to properly integrate with RPC's `preTokenBalances` and `postTokenBalances`.
+
+### Token Program Registry
+
+We need the [token-list](https://github.com/solana-labs/token-list) to include
+vetted SPL token-conforming programs.
+
+TODO consider adding Rust and C versions of the registry for on-chain programs and RPC
+
+TODO consider detailing the process of including a new program
+
+### Runtime
+
+#### Ephemeral Accounts
+
+In order to reduce friction when using read-only instructions on other programs,
+we need the ability for on-chain programs to dynamically create "ephemeral
+accounts", which only contain data, can be passed through CPI to other programs,
+and disappear after the instruction.
+
+For a program using i-tokens, to get the balance of a holding, the program creates
+an ephemeral account with 8 bytes of data, passes it to the i-token program's
+`GetHoldingBalance` instruction, then reads the information back as a `u64`. At the
+end of the instruction, the account disappears.
+
+TODO What's the best approach to implement this?  For example, does an ephemeral
+account even need a public key?
+
+This solution avoids the need to pass additional scratch accounts to almost all
+programs that use tokens.
+
+TODO what limits do we consider for these? A certain number of accounts?
+A total amount of bytes? Regarding security, this breaks the previous
+check that a CPI must use a subset of accounts provided to the calling program.
+
+#### Dynamic sysvars
+
+The i-token program also requires the use of dynamic sysvars. For example,
+during the `GetHoldingBalance` instruction, the program needs to know the current
+time in order to properly convert from token amount to share amount.
+
+Without dynamic sysvars, we need to create a system to tell what sysvars are
+needed for different instructions on each program, which creates complications
+for all users.
+
+### RPC
+
+The validator's RPC server needs to properly handle new token program accounts
+for read-only instructions and pre / post balances.
+
+#### `getBalance` and `getSupply`
+
+Instead of deserializing SPL token accounts and returning the data, the
+RPC server runs read-only transactions on the i-token program.
+
+RPC simply uses the same flow as on-chain programs: create an ephemeral account
+then runs the read-only transaction onto the bank requested.
+
+TODO how do we set the cost of read-only transactions? Do we lower it specially
```

Comment:
Maybe, this could just be free? I think we're already doing kind of read-only transactions via the pre-flight mechanism.

Reply:
Sorry, this was unclear from my side.  I was referencing compute cost here.  The transaction should definitely be free, just like a pre-flight transaction.  I was wondering if we should reduce the compute cost from 200k to something lower, since read-only instructions should be faster.

Reply:
I see. thanks for clarification. I think this can be undecided for now. More proper general fee structure will be rolled out like tx-wide compute budget (cc: @jackcmay )